autopush/websocket.py

```diff
@@ -352,13 +360,31 @@ def finish_hello(self, result):
             self.sendMessage(json.dumps(msg).encode('utf8'), False)
             return
 
+        if result.get("Attributes", {}).get("node_id"):
+            # Get the previous information returned from dynamodb.
+            attrs = result.get("Attributes")
+            node_id = self._get_aval(attrs.get("node_id", {}), 'S')
+            uaid = self._get_aval(attrs.get("uaid", {}), 'S')
+            connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
+            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
+            if not getattr(self, 'testing', False):
+                d = self.ap_settings.agent.request(
+                    "DELETE",
+                    url.encode("utf8"),
+                ).addCallback(self.finish_hello)
```

Comment:
I'd prefer to do that, frankly, but I thought that we need to defer this as a blocking call and preserve the callback chain. If I understand (quite possibly incorrectly), calling finish_hello directly would cause the DELETE to go out before we process. It seems cleaner to me to request the delete now.

Is that correct? Am I being too simplistic?